cairo/ethereum/cancun/main.cairo

```diff
@@ -59,22 +61,28 @@ func main{
     let parent_header = chain.value.blocks.value.data[
         chain.value.blocks.value.len - 1
     ].value.header;
-    let pre_state_root = parent_header.value.state_root;
-    state_transition{chain=chain}(block);
+    let (override_keccak_ptr_start) = alloc();
```

Comment:
why not just `keccak_ptr` ?

Reply:
> Main doit contenir tous les builtins du layout
Et on peut pas rebind keccak_ptr: KeccakBuiltin* en felt*

Reply:
I think you can just put `keccak_ptr: felt*` in the main signature

Reply:
it yields: 
```
    assert main_args == expected_builtin_ptrs, (
AssertionError: Expected main to contain the following arguments (in this order): ['output_ptr', 'pedersen_ptr', 'range_check_ptr', 'ecdsa_ptr', 'bitwise_ptr', 'ec_op_ptr', 'keccak_ptr', 'poseidon_ptr', 'range_check96_ptr', 'add_mod_ptr', 'mul_mod_ptr']. Found: ['output_ptr', 'pedersen_ptr', 'range_check_ptr', 'ecdsa_ptr', 'bitwise_ptr', 'ec_op_ptr', 'poseidon_ptr', 'range_check96_ptr', 'add_mod_ptr', 'mul_mod_ptr'].
keth
```

Reply:
keep the keccak_ptr, just type it `felt*` and override in the body `let (kecack_pt) = alloc()` so just no need to change the name and to pass the implicit explicitly